utils/DSen2Net.py

```diff
@@ -42,35 +47,139 @@ def init(input_shape):
     return model
 
 
+def srcnn(input_shape):
+    f1 = 9
+    f2 = 1
+    f3 = 5
+
+    n1 = 64
+    n2 = 32
+    x, _input, channels = init(input_shape)
+
+    x = Conv2D(n1, (f1, f1), activation="relu", padding="same", name="level1")(x)
+    x = Conv2D(n2, (f2, f2), activation="relu", padding="same", name="level2")(x)
+
+    out = Conv2D(channels, (f3, f3), padding="same", name="output")(x)
+
+    model = Model(inputs=_input, outputs=out)
+    return model
+
+
+def rednetsr(input_shape):
+    def _build_layer_list(model):
+        model_outputs = [layer.output for layer in model.layers]
+        return model_outputs
+
+    n_conv_layers = 15
+    n_deconv_layers = 15
+    n_skip = 2
+    n = 32
+
+    x, _input, channels = init(input_shape)
+
+    for i in range(n_conv_layers):
+        conv_idx = i + 1
+        if conv_idx == 1:
+            conv = Conv2D(n, (3, 3), activation="relu", padding="same")(x)
+        else:
+            conv = Conv2D(n, (3, 3), activation="relu", padding="same")(conv)
+
+    encoded = conv
+    encoder = Model(inputs=_input, outputs=encoded, name="encoder")
+    # Create encoder layer and output lists
+    encoder_outputs = _build_layer_list(encoder)
+
+    # CREATE AUTOENCODER MODEL
+    for i, skip in enumerate(reversed(encoder_outputs[len(_input) + 1 :])):
+
+        deconv_idx = i + 1
+        deconv_filters = n
+        if deconv_idx == n_deconv_layers:
+            deconv_filters = channels
+
+        if deconv_idx == 1:
+            deconv = Conv2DTranspose(
+                deconv_filters, (3, 3), activation="relu", padding="same"
+            )(encoded)
+        else:
+            deconv = Conv2DTranspose(
+                deconv_filters, (3, 3), activation="relu", padding="same"
+            )(deconv)
+
+        if deconv_idx % n_skip == 0:
+            deconv = Add()([deconv, skip])
+            ReLU()(deconv)
+
+    decoded = deconv  # (decoder_inputs)
+    model = Model(inputs=_input, outputs=decoded)
+    return model
+
+
+def resnetsr(input_shape):
+    def _residual_block(ip, _id):
+        channel_axis = 1  # channels first
+        init = ip
+
+        x = Conv2D(n, (3, 3), padding="same", name="sr_res_conv_" + str(_id) + "_1")(ip)
+        x = BatchNormalization(
+            momentum=0.5, axis=channel_axis, name="sr_res_batchnorm_" + str(_id) + "_1"
+        )(x)
+        x = PReLU(
+            alpha_initializer="zeros",
+            alpha_regularizer=None,
+            alpha_constraint=None,
+            shared_axes=[2, 3],
+        )(x)
+        x = Conv2D(n, (3, 3), padding="same", name="sr_res_conv_" + str(_id) + "_2")(x)
+        x = BatchNormalization(
+            momentum=0.5, axis=channel_axis, name="sr_res_batchnorm_" + str(_id) + "_2"
+        )(x)
+
+        m = Add(name="sr_res_merge_" + str(_id))([x, init])
+        return m
+
+    n = 64
+    x, _input, channels = init(input_shape)
+
+    x0 = Conv2D(n, (9, 9), padding="same", name="sr_res_conv1")(x)
+    x0 = PReLU(
```

Comment:
Do we use PReLU only in this model or in all blocks as activation function?
I wonder if it is worthwhile using the same one overall. On the downside: it is possible that this choice has effects on runtime as well as convergence.

Reply:
PReLu seems to be only used in the ResNetSR model,  the orginal model uses a ReLU activation, as well as most of the other models. Should I change this?

Reply:
Maybe we stick with ReLU overall so the competition is "fair"?

Reply:
Sure. makes sense! Only ResNetSR will change then.